src/node_api.cc

```diff
@@ -557,65 +576,67 @@ void napi_module_register(napi_module* mod) {
   node::node_module_register(nm);
 }
 
-#define RETURN_STATUS_IF_FALSE(condition, status)                       \
+#define RETURN_STATUS_IF_FALSE(env, condition, status)                  \
   do {                                                                  \
     if (!(condition)) {                                                 \
-      return napi_set_last_error((status));                             \
+      return napi_set_last_error((env), (status));                      \
     }                                                                   \
   } while (0)
 
-#define CHECK_ARG(arg) RETURN_STATUS_IF_FALSE((arg), napi_invalid_arg)
+#define CHECK_ENV(env) \
+  assert((env) != nullptr)
```

Comment:
It doesn't necessarily mean that even the native module is in an invalid state (at least not any more than any other NAPI error does). It could be possible for other calls to the native module to succeed.